src/routes/download/download.router.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response, Router } from 'express';
+import { get } from 'lodash';
+import JSZip from 'jszip';
+import moment from 'moment';
+
+import { DOWNLOAD_SEQUENCES_ENDPOINT } from '../../constants/endpoint';
+import { downloadSequenceFiles } from './download.service';
+import logger from '../../logger';
+
+export const router: Router = Router();
+
+router.post(DOWNLOAD_SEQUENCES_ENDPOINT, async (req: Request, res: Response) => {
+  const ids = get(req.body, 'ids', []);
+  console.time('zip download');
+  try {
+    const assets = await downloadSequenceFiles(ids);
+    const zip = new JSZip();
+    await assets.map(async (asset) => {
+      const sequenceId = (await asset).sequenceId;
+      // writing the files to a separate folder for each sequence id
+      // if there are no files for an individual id, the folder is created but will be empty so the user is aware
+      zip.folder(sequenceId);
+      await Promise.all(
+        asset.files.map((file) => {
+          if (file) {
+            logger.info(`Writing file ${file.name} to zip folder ${sequenceId}`);
+            zip.file(`${sequenceId}/${file.name}`, file.stream);
+          }
+        }),
+      );
+    });
```

Comment:
This work should be performed in the download service as it is unrelated to the router. Handling the zip stream and res setting should occur here but none of the work for building the payload.

Also, I see all the work in building the zip as being a method on its own `zipSequenceFiles(assets)`.